rosserial_arduino/src/ros_lib/ArduinoHardware.h

```diff
@@ -49,14 +49,15 @@
     #define SERIAL_CLASS usb_serial_class
   #endif
 #elif defined(_SAM3XA_)
-  #include <UARTClass.h>  // Arduino Due
-  #define SERIAL_CLASS UARTClass
+  #if defined(USE_SERIALUSB)
+    #define SERIAL_CLASS Serial_
+  #else
+    #include <UARTClass.h>  // Arduino Due
+    #define SERIAL_CLASS UARTClass
+  #endif
 #elif defined(USE_USBCON)
   // Arduino Leonardo USB Serial Port
   #define SERIAL_CLASS Serial_
-#elif (defined(__STM32F1__) and !(defined(USE_STM32_HW_SERIAL))) or defined(SPARK) 
-  // Stm32duino Maple mini USB Serial Port
```

Comment:
Why do you delete every STM32 entries?
You seem to use a version prior to https://github.com/ros-drivers/rosserial/commit/ce24e604db1cbab069332e183d42de4c5784d62f